test/bigtable/index.js

```diff
@@ -0,0 +1,313 @@
+  });
+
+  beforeEach(function() {
+    bigtable = new Bigtable({
+      projectId: PROJECT_ID,
+      zone: ZONE,
+      cluster: CLUSTER
+    });
+  });
+
+  describe('instantiation', function() {
+    it('should normalize the arguments', function() {
+      var normalizeArguments = fakeUtil.normalizeArguments;
+      var normalizeArgumentsCalled = false;
+      var fakeOptions = {
+        projectId: PROJECT_ID,
+        zone: ZONE,
+        cluster: CLUSTER
+      };
+      var fakeContext = {};
+
+      fakeUtil.normalizeArguments = function(context, options) {
+        normalizeArgumentsCalled = true;
+        assert.strictEqual(context, fakeContext);
+        assert.strictEqual(options, fakeOptions);
+        return options;
+      };
+
+      Bigtable.call(fakeContext, fakeOptions);
+      assert(normalizeArgumentsCalled);
+
+      fakeUtil.normalizeArguments = normalizeArguments;
+    });
+
+    it('should localize the cluster name', function() {
+      assert(bigtable.clusterName, CLUSTER_NAME);
+    });
+
+    it('should inherit from GrpcService', function() {
+      assert(bigtable instanceof GrpcService);
+
+      var calledWith = bigtable.calledWith_[0];
+
+      assert.strictEqual(calledWith.baseUrl, 'bigtable.googleapis.com');
+      assert.strictEqual(calledWith.service, 'bigtable');
+      assert.strictEqual(calledWith.apiVersion, 'v1');
+
+      assert.deepEqual(calledWith.protoServices, {
+        BigtableService: googleProtoFiles.bigtable.v1,
+        BigtableTableService: {
+          path: googleProtoFiles.bigtable.admin,
+          service: 'bigtable.admin.table'
+        }
+      });
+
+      assert.deepEqual(calledWith.scopes, [
+        'https://www.googleapis.com/auth/bigtable.admin',
+        'https://www.googleapis.com/auth/bigtable.data'
+      ]);
+    });
+  });
+
+  describe('formatTableName_', function() {
+    it('should return the last section of a formatted table name', function() {
+      var fakeTableName = 'projects/p/zones/z/clusters/c/tables/my-table';
+      var formatted = Bigtable.formatTableName_(fakeTableName);
+
+      assert(formatted, 'my-table');
+    });
+
+    it('should do nothing if the table is name is not formatted', function() {
+      var fakeTableName = 'my-table';
+      var formatted = Bigtable.formatTableName_(fakeTableName);
+
+      assert(formatted, fakeTableName);
+    });
+  });
+
+  describe('createTable', function() {
+    var TABLE_ID = 'my-table';
+
+    it('should provide the proper request options', function(done) {
+      bigtable.request = function(protoOpts, reqOpts) {
+        assert.deepEqual(protoOpts, {
+          service: 'BigtableTableService',
+          method: 'createTable'
+        });
+
+        assert.strictEqual(reqOpts.name, CLUSTER_NAME);
+        assert.strictEqual(reqOpts.tableId, TABLE_ID);
+        done();
+      };
+
+      bigtable.createTable(TABLE_ID, assert.ifError);
+    });
+
+    it('should not modify the original options', function(done) {
```

Comment:
I don't this this test is necessary. Maybe if we provided real arguments (`options.operation` and `options.splits`), then tested we didn't change the options object it would be more useful. WDYT?